csharp/src/Apache.Arrow/Arrays/ListArray.cs

```diff
@@ -77,7 +77,7 @@ public Builder AppendNull()
 
             public ListArray Build(MemoryAllocator allocator = default)
             {
-                Append();
+                ValueOffsetsBufferBuilder.Append(ValueBuilder.Length);
```

Comment:
Can you write a unit test for this to ensure it stays fixed?

Reply:
OK, I have added a test for this.